include/nbl/builtin/hlsl/scan/descriptors.hlsl

```diff
@@ -1,3 +1,112 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
 
+#ifndef _NBL_HLSL_SCAN_DESCRIPTORS_INCLUDED_
+#define _NBL_HLSL_SCAN_DESCRIPTORS_INCLUDED_
 
-// choerent -> globallycoherent
\ No newline at end of file
+#include "nbl/builtin/hlsl/scan/declarations.hlsl"
+#include "nbl/builtin/hlsl/workgroup/basic.hlsl"
+
+// coherent -> globallycoherent
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace scan
+{
+
+template<uint32_t scratchElementCount=scratchSz> // (REVIEW): This should be externally defined. Maybe change the scratch buffer to RWByteAddressBuffer? Annoying to manage though...
+struct Scratch
+{
+    uint32_t workgroupsStarted;
+    uint32_t data[scratchElementCount];
+};
+
+[[vk::binding(0 ,0)]] RWStructuredBuffer<uint32_t /*Storage_t*/> scanBuffer; // (REVIEW): Make the type externalizable. Decide how (#define?)
+[[vk::binding(1 ,0)]] RWStructuredBuffer<Scratch> globallycoherent scanScratchBuf; // (REVIEW): Check if globallycoherent can be used with Vulkan Mem Model
```

Comment:
use BDA for the scratch and pass it in as a `uint64_t` in the Scan parameter

to get coherent behaviour you can try using a spir-v OpLoad/OpStore/OpAtomicUAdd with the correct VMM operands, but not sure if you'll get pwned by DXC's crappy BDA support, so do some godbolting first to check.